core/modules/ccontrol/UserQueryResources.h

```diff
@@ -0,0 +1,118 @@
+ * Copyright 2019 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+
+#ifndef LSST_QSERV_CCONTROL_USERQUERYRESOURCES_H
+#define LSST_QSERV_CCONTROL_USERQUERYRESOURCES_H
+
+
+#include <memory>
+#include <string>
+
+#include "qmeta/types.h"
+#include "mysql/MySqlConfig.h"
+
+
+namespace lsst {
+namespace qserv {
+namespace ccontrol {
+    class UserQueryResources;
+}
+namespace css {
+    class CssAccess;
+}
+namespace mysql {
+    class MySqlConfig;
+}
+namespace qmeta {
+    class QMeta;
+    class QStatus;
+    class QMetaSelect;
+}
+namespace qproc {
+    class SecondaryIndex;
+}
+namespace sql {
+    class SqlConnection;
+}}}
+
+
+namespace lsst {
+namespace qserv {
+namespace ccontrol {
+
+
+/**
+ * UserQueryResources owns the Czar resources that are useful to & shared among UserQuery instances.
+ */
+class UserQuerySharedResources {
+
+public:
+    UserQuerySharedResources(std::shared_ptr<css::CssAccess> css_,
+                             mysql::MySqlConfig const& mysqlResultConfig_,
+                             std::shared_ptr<qproc::SecondaryIndex> secondaryIndex_,
+                             std::shared_ptr<qmeta::QMeta> queryMetadata_,
+                             std::shared_ptr<qmeta::QStatus> queryStatsData_,
+                             std::shared_ptr<qmeta::QMetaSelect> qMetaSelect_,
+                             std::shared_ptr<sql::SqlConnection> resultDbConn_,
+                             std::string const& czarName);
+
+    UserQuerySharedResources(UserQuerySharedResources const& rhs) = default;
+    UserQuerySharedResources& operator=(UserQuerySharedResources const& rhs) = delete;
+
+    std::shared_ptr<css::CssAccess> css;
+    mysql::MySqlConfig const mysqlResultConfig;
+    std::shared_ptr<qproc::SecondaryIndex> secondaryIndex;
+    std::shared_ptr<qmeta::QMeta> queryMetadata;
+    std::shared_ptr<qmeta::QStatus> queryStatsData;
+    std::shared_ptr<qmeta::QMetaSelect> qMetaSelect;
+    std::shared_ptr<sql::SqlConnection> resultDbConn;
+    qmeta::CzarId qMetaCzarId;   ///< Czar ID in QMeta database
+
+    std::shared_ptr<UserQueryResources> makeUserQueryResources(std::string const& userQueryId, std::string const& resultDb);
```

Comment:
Does this method do anything different from calling public UserQueryResources constructor?

Reply:
yes, I'll add documentation.